### PR TITLE
MODFQMMGR-3 Use more specific FQM interfaces

### PR DESCRIPTION
**Pull request by mweaver-ebsco**
This commit changes the module's declared dependencies in the module descriptor from "mod-fqm-manager" to the more specific "entity-types" and "query" as the old one is going away and we need to use the more specific ones anyway

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -2,11 +2,6 @@
   "id": "@artifactId@-@version@",
   "name": "The module descriptor for mod-lists",
   "provides": [
-    {
-      "id": "mod-lists",
-      "version": "1.0",
-      "handlers": []
-    },
```
Comment (mweaver-ebsco) [inline descriptors/ModuleDescriptor-template.json:9]: Whoops, I meant to put this part in a different commit...

**Review by bvsharp — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -154,7 +149,11 @@
   ],
   "requires": [
     {
-      "id": "mod-fqm-manager",
+      "id": "query",
```
Comment (craigmcnally) [inline descriptors/ModuleDescriptor-template.json:152]: `query` is a tad generic/broad.  Consider having mod-fqm-manager provider `fqm` or `fqm-query`.

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -2,11 +2,6 @@
   "id": "@artifactId@-@version@",
   "name": "The module descriptor for mod-lists",
   "provides": [
-    {
-      "id": "mod-lists",
```
Comment (craigmcnally) [inline descriptors/ModuleDescriptor-template.json:6]: `lists` is a much better interface name than `mod-lists`.

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (mweaver-ebsco) [reply descriptors/ModuleDescriptor-template.json]: Good call! I went ahead and renamed it to `fqm-query`
